scripts/run-tests.sh

```diff
@@ -112,29 +115,32 @@ mkdir -p ${WORKSPACE}
 
 get_pytest_files
 
+test_result=0
+
 convert_and_test "qemux86_64" \
                  "release-1" \
                  "${UBUNTU_IMAGE_URL}" \
                  "${UBUNTU_IMAGE}" \
                  "${UBUNTU_IMAGE}.gz" \
-                 "configs/qemux86-64_config"
-
+                 "configs/qemux86-64_config" || test_result=$?
 
 convert_and_test "raspberrypi" \
                  "release-1" \
                  "${RASPBIAN_IMAGE_URL}" \
                  "${RASPBIAN_IMAGE}.img" \
                  "${RASPBIAN_IMAGE}.zip" \
-                 "configs/raspberrypi3_config"
+                 "configs/raspberrypi3_config" || test_result=$?
 
 convert_and_test "linaro-alip" \
                  "release-1" \
                  "${TINKER_IMAGE_URL}" \
                  "${TINKER_IMAGE}.img" \
-                 "${TINKER_IMAGE}.zip"
+                 "${TINKER_IMAGE}.zip" || test_result=$?
 
 convert_and_test "beaglebone" \
                  "release-1" \
                  "${BBB_DEBIAN_IMAGE_URL}" \
                  "${BBB_DEBIAN_IMAGE}" \
-                 "${BBB_DEBIAN_IMAGE}.xz"
+                 "${BBB_DEBIAN_IMAGE}.xz" || test_result=$?
```

Comment:
This will not work right? If the final test succeeds but all previous have failed, this would still set the test_result to 0.

I think we need to OR the return values, to keep the `1`

Reply:
Oh, nice catch! I missed that!

Reply:
It works: it only sets test_result to the exit code of the previous command if the command fails. On success it does nothing

I have tested it on the CI and locally.

If more than one failed, yes, you will be getting only the last errcode, but that is okey I think.

Reply:
Thanks for clarifying. Makes sense.